WordPress/Classes/ViewRelated/Post/Revisions/RevisionsTableViewController.swift

```diff
@@ -14,50 +26,133 @@ class RevisionsTableViewController: UITableViewController {
+        fetchRequest.predicate = predicate
+        fetchRequest.sortDescriptors = [descriptor]
+        return fetchRequest
     }
 
-    func fetchRevisions() {
-        guard let postRevisions = post.revisions as? [Int] else {
-            return
+    func sectionNameKeyPath() -> String {
+        return #keyPath(Revision.revisionDate)
+    }
+
+    func configureCell(_ cell: UITableViewCell, at indexPath: IndexPath) {
+        guard let cell = cell as? RevisionsTableViewCell else {
+            preconditionFailure("The cell should be of class \(String(describing: RevisionsTableViewCell.self))")
+        }
+
+        let revision = self.revision(at: indexPath)
+        cell.revisionNum = revision.revisionId.intValue
+    }
+
+    func revision(at indexPath: IndexPath) -> Revision {
+        guard let revision = tableViewHandler.resultsController.object(at: indexPath) as? Revision else {
+            fatalError("Expected a Revision object.")
         }
-        revisions = postRevisions
+
+        return revision
+    }
+
+
+    // MARK: Override delegate methodds
+
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return Constant.Size.sectionHeaderHeight
+    }
+
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return 0.0
     }
 
-    func numberOfSections(in tableView: UITableView) -> Int {
-        return 1
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return Constant.Size.cellEstimatedRowHeight
     }
 
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return revisions.count
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let nibName = String(describing: PageListSectionHeaderView.self)
+        guard let sectionInfo = tableViewHandler.resultsController.sections?[section],
+            let headerView = Bundle.main.loadNibNamed(nibName, owner: nil, options: nil)?.first as? PageListSectionHeaderView else {
+                return UIView(frame: .zero)
+        }
+
+        headerView.setTite(sectionInfo.name)
+        return headerView
     }
 
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: RevisionsTableViewCell.reuseIdentifier) as? RevisionsTableViewCell else {
-            return UITableViewCell()
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: RevisionsTableViewCell.reuseIdentifier, for: indexPath) as? RevisionsTableViewCell else {
+            preconditionFailure("The cell should be of class \(String(describing: RevisionsTableViewCell.self))")
```

Comment:
Because it's an approach I don't really like it. Trying to recover at runtime just mask the problem. Returning an empty cell doesn't fix the cell registration problem (the only reason why a cell shouldn't be of the type I expect) but adds code we have to handle in other places later (for instance within a delegate method). Swift uses a lot of exceptions and are really helpful. Always talking about the *tableview*, there are helpers using generics we can write in order to avoid the *guard let* duplications, like:
```
func cell<T: UITableViewCell>(for tableView: UITableView, at indexPath: IndexPath, identifier: String) -> T {
        guard let cell = tableView.dequeueReusableCell(withIdentifier: identifier) as? T else {
            fatalError("A cell must be found for identifier: \(identifier)")
        }

        return cell
    }
```
(I also use other to simplify the registration and reusableview identifiers)
If our approach is to return the empty cell I'll do it but I would like to avoid adding patch to mask problem caused somewhere else 😊.
Let me know what you think

Reply:
I personally don't like crashing in production. I prefer `assertionFailure` since it will crash in debug builds but not prod.

We don't have a guideline covering this—tho I wish we did 😜